src/components/portal/portal.tsx

```diff
@@ -252,6 +251,21 @@ export class Portal {
         this.container.classList.add(IS_VISIBLE_CLASS);
     }
 
+    private animateHideAndCleanup() {
+        if (!this.container) {
+            return;
+        }
+
+        this.container.classList.add(IS_HIDING_CLASS);
+        this.styleContainer();
+
+        setTimeout(() => {
+            this.container.classList.remove(IS_HIDING_CLASS);
+            this.container.classList.remove(IS_VISIBLE_CLASS);
+            this.destroyPopper();
+        }, hideAnimationDuration);
```

Comment:
These timeouts can prove tricky. I'm not sure if this is a valid use case and what should be expected. But if you click it three times it ends up in a strange state of being opened but it's now showing.

https://github.com/user-attachments/assets/bc359aad-a44a-4725-bd5c-1e4d04812131

Reply:
Just to be clear, I'm not sure we should consider this scenario.

Reply:
good point. 🤔 but we need the timeout for letting the animations run. Is there a better way to do this?

Maybe I can come up with a more "native" way, using this: https://developer.mozilla.org/en-US/docs/Web/CSS/transition-behavior

But I have to check it again. It can be a bit tricky sometimes.

Reply:
Oh I actually can't. I tried this before, but ended up adding this timeout. The thing is that I need to keep those dynamically applied inlined styles on the container while the animation is running. The javascript removes them and everything gets meaningless.

Reply:
I tied to fix it using a state variable. But I couldn't.